server/aesdsocket.c

```diff
@@ -350,6 +368,33 @@ void *threadfunc(void *args)
                 syslog(LOG_ERR, "Error Msg: %s", strerror(errno));
                 closeAll(EXIT_FAILURE);
             }
+
+        #if (USE_AESD_CHAR_DEVICE == 1)
+        if(strncmp(my_buffer, seek_string, strlen(seek_string)) == 0)
+        {
+            struct aesd_seekto seekto;
+            syslog(LOG_DEBUG, "Write command to seek.");
+            if(sscanf(my_buffer, "AESDCHAR_IOCSEEKTO:%u,%u", &seekto.write_cmd, &seekto.write_cmd_offset) != 2)
+            {
+                syslog(LOG_ERR, "sscanf() fail");
+                syslog(LOG_ERR, "sscanf() call err. error string is %s", strerror(errno));
+            }
+            syslog(LOG_DEBUG, "seekto.write_cmd is %u and seekto.write_cmd_offset is %u", seekto.write_cmd, seekto.write_cmd_offset);
+            int result_ret = ioctl(recvfile_fd, AESDCHAR_IOCSEEKTO, &seekto);
+            if (result_ret == 0)
+            {
+                syslog(LOG_DEBUG, "ioctl() call successful");
+            }
+            else if (result_ret == -1)
+            {
+                syslog(LOG_ERR, "ioctl() call err. error string is %s", strerror(errno));
+            }
+
+            goto skip_write;
```

Comment:
I don't think you want to exit here... I think you are skipping over the logic to send the content back over the socket at the seek to offset.  This is probably why https://github.com/cu-ecen-aeld/assignment-9-sota6640/actions/runs/11699177513/job/32580691116 is failing.